cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -159,18 +160,33 @@ func (a *Actuator) deleteAsyncEmpty(NodeGroupViews []*budgets.NodeGroupView) (re
 func (a *Actuator) taintNodesSync(NodeGroupViews []*budgets.NodeGroupView) errors.AutoscalerError {
 	var taintedNodes []*apiv1.Node
 	for _, bucket := range NodeGroupViews {
+		startTimestamp := map[string]time.Time{}
+		updateLatencyTracker := NewUpdateLatencyTracker(startTimestamp, a.ctx.AutoscalingKubeClients.ListerRegistry.AllNodeLister())
+		stopChan := make(chan struct{})
+		resultChan := make(chan int64)
+		go updateLatencyTracker.Start(stopChan, resultChan)
 		for _, node := range bucket.Nodes {
+			startTimestamp[node.Name] = time.Now()
 			err := a.taintNode(node)
 			if err != nil {
 				a.ctx.Recorder.Eventf(node, apiv1.EventTypeWarning, "ScaleDownFailed", "failed to mark the node as toBeDeleted/unschedulable: %v", err)
 				// Clean up already applied taints in case of issues.
 				for _, taintedNode := range taintedNodes {
 					_, _ = taints.CleanToBeDeleted(taintedNode, a.ctx.ClientSet, a.ctx.CordonNodeBeforeTerminate)
 				}
+				close(stopChan)
 				return errors.NewAutoscalerError(errors.ApiCallError, "couldn't taint node %q with ToBeDeleted", node)
 			}
 			taintedNodes = append(taintedNodes, node)
 		}
+		close(stopChan)
+		latency := <-resultChan
+		latencyDuration, err := time.ParseDuration(fmt.Sprintf("%vms", latency))
```

Comment:
Woot, why convert int to a string, then parse it back to int (i.e. Duration) again?

Reply:
Removed that part